src/main/public/js/postcode-lookup.js

```diff
@@ -18,9 +18,32 @@
             hide(postcodeAddressPicker(postcodeLookupWidget))
             clearAddressFields(postcodeLookupWidget)
             hide(addressSection(postcodeLookupWidget))
-            lookupPostcode(this.previousElementSibling.value, postcodeLookupWidget)
+            var enteredPostcode = this.previousElementSibling.value;
+
+            if (isUKPostcode(enteredPostcode) && !isScotlandOrWalesPostcode(enteredPostcode)) {
+              lookupPostcode(enteredPostcode, postcodeLookupWidget);
+            } else {
+              displayPostcodeError(postcodeLookupWidget);
+            }
           })
 
+        function isUKPostcode(postcode) {
+          var ukPostcodePattern = /^[A-Z]{1,2}\d[A-Z\d]? \d[A-Z]{2}$/i;
```

Comment:
I think this is the regex for UK postcode validation: `/^(([Gg][Ii][Rr] 0[Aa]{2})|((([A-Za-z][0-9]{1,2})|(([A-Za-z][A-Ha-hJ-Yj-y][0-9]{1,2})|(([A-Za-z][0-9][A-Za-z])|([A-Za-z][A-Ha-hJ-Yj-y][0-9]?[A-Za-z]))))[0-9][A-Za-z]{2}))$/ `

Reply:
I check that in documents you send to me - thank you for that